src/style/style.js

```diff
@@ -385,6 +391,21 @@ class Style extends Evented {
         return false;
     }
 
+    get order(): Array<string> {
+        if (this.map._optimizeForTerrain && this.terrain) {
+            if (!this._drapedFirstOrder.length) {
+                this._updateDrapeFirstLayers();
+            }
+            return this._drapedFirstOrder;
+        }
+        return this._order;
+    }
+
+    isLayerDraped(layer: StyleLayer): boolean {
+        if (!this.terrain) return false;
+        return drapedLayers.hasOwnProperty(layer.type);
```

Comment:
I don't think you need to use `hasOwnProperty` `drapedLayers[layer.type]` should work too.

Reply:
Updated in https://github.com/mapbox/mapbox-gl-js/pull/10258/commits/b732630d4465ce232a58c71932843833dc2885e9.